packages/destination-actions/src/destinations/metronome/sendEvent/index.ts

```diff
@@ -0,0 +1,85 @@
+import type { ActionDefinition } from '@segment/actions-core'
+import type { Settings } from '../generated-types'
+import type { Payload } from './generated-types'
+
+function serializeEvent(event: Payload) {
+  return {
+    ...event,
+    // convert the timestamp into rfc3999 format
+    timestamp: new Date(event.timestamp).toISOString(),
+  }
+}
+
+const action: ActionDefinition<Settings, Payload> = {
+  title: 'Send Event',
+  description: 'Send an event to Metronome',
+  fields: {
+    transaction_id: {
+      type: 'string',
+      label: 'transaction_id',
+      description: 'The Metronome transaction ID uniquely identifies an event to ensure Metronome only processes each event once.',
+      required: true,
+      default: {
+        '@path': '$.messageId'
+      }
+    },
+    customer_id: {
+      type: 'string',
+      label: 'customer_id',
+      description: 'The Metronome customer ID or ingest alias this event should be associated with.',
+      required: true,
+      default: {
+        // By default, use the group ID if it exists, otherwise use the customer ID. If neither exist, use the anonymous ID.
+        '@if': {
+          exists: { '@path': '$.groupId' },
+          then: { '@path': '$.groupId' },
+          else: {
+            '@if': {
+              exists: { '@path': '$.userId' },
+              then: { '@path': '$.userId' },
+              else: { '@path': '$.anonymousId' }
+            }
+          }
+        }
+      }
+    },
+    timestamp: {
+      type: 'datetime',
+      label: 'timestamp',
+      description: 'The timestamp at which this event occurred.',
```

Comment:
Should the comment include that the format is RFC 3999?

Reply:
It should not - Since we mark the field as `date-time` in the action, segment will enforce that people only send "date-time" data to the field. The format of the data before being sent to our action is Segment's problem. 

There's a deeper discussion of this here. https://metronometechnologies.slack.com/archives/C02LRAEV4KU/p1637020717017000

I would expect at some point in the future Segment will update their actions code (in this repo) to pass in `Date` instances instead of `string | number`, and the cool thing about that is that the types will automatically fail, and they can safely fix the types and trust the test cases in that case (hopefully) without involving us at all.

Reply:
nevermind. i guess the serializeEvent will take care of this. But maybe something that says it will get converted to RFC3999?

Reply:
What it gets converted to doesn't really matter to the user right? As far as they know we could convert their whole event into a protobuf and send it over the wire.

Reply:
This field is in the Segment UI as "type: date" and then they have to select a field in their data that is a date type.

Reply:
yeah that makes sense. My original concern was that it would hard for clients to figure out what the converted value would be but I think this is actually a moot point. Doing a few conversions using Date() everything converts as expected with the toISOString() method.

Reply:
I switched to `dayjs` which everyone thinks should be the same as `new Date()` but segment has only promised to pass things that match what `dayjs` expects so might as well be safe. Also added a comment to this point.